protos/core.proto

```diff
@@ -289,99 +289,375 @@ enum SupportedFeatures {
   ARRAY_BUFFER_IPC = 1;
 }
 
+// Workflow Settings Config defines the contents of the `workflow_settings.yaml`
```

Comment:
nit: do you want to put these new messages in `config_files.proto` or something?